src/Conduit/Middleware/AuthenticationMiddleware.php

```diff
@@ -18,13 +18,13 @@ public function __construct(Auth $auth, ResumeService $resume_service)
         $this->auth = $auth;
     }
 
-    public function __invoke(Request $request, Response $response, callable $next = null)
+    public function __invoke(ServerRequestInterface $request, ResponseInterface $response, callable $next = null)
     {
-        if ($this->auth->isValid()) {
-            $next();
-        } else {
-            $response->setStatus(401);
-            $response->setHeader('Location', '/login');
+        if (! $this->auth->isValid()) {
+            $response = $response
+                ->withStatus(401)
+                ->withHeader('Location', '/login');
         }
+        return $next($request, $response);
```

Comment:
@simensen see I am calling the next middleware with the request and changed response. Now it will call the router middleware. The router middleware actually can route for there is a route. So this is like you need to call some sort of exception and then call something else to end up.

If you look at the earlier version it is very simple for `at-least` simple things. May be not good for complex things :) .